core/domain/suggestion_services.py

```diff
@@ -1826,3 +1828,896 @@ def update_question_suggestion(
+    user_id: str, target_id: str, suggestion_is_accepted: bool
+) -> None:
+    """Creates/updates QuestionReviewStatsModel model for given question
+    reviewer when a question is sreviewed.
+
+    Args:
+        suggestion: Suggestion. The suggestion domain object that is being
+            reviewed.
+        user_id: str. The ID of the corresponding reviewer.
+        target_id: str. The ID of the target entity being suggested to.
+        suggestion_is_accepted: bool. Indicates whether whether the suggestion
+            is accepted.
+    """
+    topic_ids = [
+        topic.topic_id
+        for topic in skill_services.get_all_topic_assignments_for_skill(
+            target_id)
+    ]
+
+    for topic_id in topic_ids:
+        question_review_stat_model = (
+            suggestion_models.QuestionReviewStatsModel.get(
+                user_id, topic_id
+            ))
+
+        if question_review_stat_model is None:
+            suggestion_models.QuestionReviewStatsModel.create(
+                reviewer_user_id=user_id,
+                topic_id=topic_id,
+                reviewed_questions_count=1,
+                accepted_questions_count=int(suggestion_is_accepted),
+                accepted_questions_with_reviewer_edits_count=int(
+                    suggestion.edited_by_reviewer),
+                first_contribution_date=_get_contribution_date(
+                    suggestion.last_updated),
+                last_contribution_date=_get_contribution_date(
+                    suggestion.last_updated)
+            )
+            continue
+
+        question_review_stat = (
+            _create_question_review_stats_from_model(
+                question_review_stat_model))
+
+        get_incremental_question_review_stats(
+            question_review_stat, suggestion.last_updated,
+            suggestion_is_accepted,
+            suggestion.edited_by_reviewer)
+
+        _update_question_review_stats_models([question_review_stat])
+
+    update_question_contribution_stats_at_review(
+        suggestion, suggestion.author_id, target_id)
+
+
+def get_incremental_translation_contribution_stats_at_review(
+    translation_contribution_stat: (
+        suggestion_registry.TranslationContributionStats),
+    content_word_count: int,
+    suggestion_is_accepted: bool,
+    edited_by_reviewer: bool
+) -> None:
+    """Updates TranslationContributionStats object.
+
+    Args:
+        translation_contribution_stat: TranslationContributionStats. The stats
+            object to update.
+        content_word_count: int. The number of words in the translation.
+        suggestion_is_accepted: bool. A flag that indicates whether the
+            suggestion is accepted.
+        edited_by_reviewer: bool. A flag that indicates whether the suggestion
+            is edited by the reviewer.
+    """
+    (
+        translation_contribution_stat
+        .accepted_translations_count
+    ) += int(suggestion_is_accepted)
```

Comment:
This can fit one line or can be split up so that it is more readable (after `int(` for example).

Reply:
Done